crates/router/src/core/payments.rs

```diff
@@ -155,11 +143,12 @@ where
                     state,
                     &merchant_account,
                     connector,
-                    &operation,
+                    operation.to_update_tracker()?,
```

Comment:
send whole operation itself. call_connector_service is not just specific to just one operation.